pkg/runtime/optimize/optimizer.go

```diff
@@ -693,6 +695,12 @@ func (o optimizer) optimizeShowIndex(_ context.Context, stmt *rast.ShowIndex, ar
 	return ret, nil
 }
 
+func (o optimizer) optimizeShowColumns(_ context.Context, stmt *rast.ShowColumns, args []interface{}) (proto.Plan, error) {
+	ret := &plan.ShowColumnsPlan{Stmt: stmt}
+	ret.BindArgs(args)
+	return ret, nil
+}
```

Comment:
Should compute shard route, because the target table may be a logical table. For example, the statement of `show columns from student`, here the target `student` is a logical table, you should route it to `employees_0000.student_0000`, and submit `show columns from student_0000`.

Reply:
Did it